### ISSUE TITLE
Exit with error if a resource has no files in Python script

### ISSUE THREAD

**Issue by clapierre** [benetech/ImageShare#156]
I am having to duplicate each row in our excel spreadsheet for the DCMP videos where row 1 has the resource information and row 2 has the file information for that resource.
If I miss duplicating a row the result will be a resource without any files associated with it which is not good.  I would rather get an error message telling me which resource_unique_name has no files associated with it, so I can fix the Excel / Text file.

@sinabahram would you be able to modify the python script to abort with that error message?

**Comment by clapierre** [comment]
Or even better would be not having to duplicate each row for resource/file when there is only one file associated with a resource and use the file_* fields on the resource row to fill in the resource file, but I am sure that is more work.  Just means a lot of copy/insert new copied row for all the videos.

**Comment by sinabahram** [comment]
I would really like to avoid changing the data model at this point. It’s been pretty well-tested across several iterations.

 

As for an error when there are no files for a resource, that’ snot an error. That’s a perfectly ok state of data, so why should we throw an error?

**Comment by clapierre** [comment]
This was with creating of the JSON file, but is now less important with my new script which duplicates each row with duplicaterows.py in the Scripts folder under imageshare-utils.

The issue is Sina I get data with 1 record containing both resource and its single file in one row.
and I had to go in to excel and manually duplicate 1000 rows of data so that we have the resource on one row and file associated with that resource on the second row.  and if I messed up and forgot to duplicate a row then I ended up with a resource without any files associated with it.

So I have now created this script and for those libraries we receive that only have a single file with a resource I run my script and it duplicates each row after the headers and then I can run your tsv2json.py script to convert that into a JSON file which has now each resource and its single resource file (due the the duplicate row being added for each record by my script, since the data was already there just needed to be added on the 2nd row.

I can close this now as this is pretty much handled.
